lib/common/alarm/AlarmFactory.ts

```diff
@@ -496,20 +517,62 @@ export class AlarmFactory {
       );
     }
 
-    const alarm = adjustedMetric.createAlarm(this.alarmScope, alarmName, {
+    // create primary alarm
+
+    const primaryAlarm = adjustedMetric.createAlarm(
+      this.alarmScope,
       alarmName,
-      alarmDescription,
-      threshold: props.threshold,
-      comparisonOperator: props.comparisonOperator,
-      treatMissingData: props.treatMissingData,
-      // default value (undefined) means "evaluate"
-      evaluateLowSampleCountPercentile: evaluateLowSampleCountPercentile
-        ? undefined
-        : "ignore",
-      datapointsToAlarm,
-      evaluationPeriods,
-      actionsEnabled,
-    });
+      {
+        alarmName,
+        alarmDescription,
+        threshold: props.threshold,
+        comparisonOperator: props.comparisonOperator,
+        treatMissingData: props.treatMissingData,
+        // default value (undefined) means "evaluate"
+        evaluateLowSampleCountPercentile: evaluateLowSampleCountPercentile
+          ? undefined
+          : "ignore",
+        datapointsToAlarm,
+        evaluationPeriods,
+        actionsEnabled,
+      }
+    );
+
+    let alarm: AlarmBase = primaryAlarm;
+
+    // create composite alarm for min metric samples (if defined)
+
+    if (props.minMetricSamplesToAlarm) {
+      const metricSampleCount = metric.with({ statistic: MetricStatistic.N });
+      const noSamplesAlarm = metricSampleCount.createAlarm(
+        this.alarmScope,
+        alarmName + "-NoSamples",
```

Comment:
```suggestion
        `${alarmName}-NoSamples`,
```

Reply:
Likewise for other string concatenations.

Reply:
Will fix